TestEnforcingServiceAccount flake

```
--- FAIL: TestEnforcingServiceAccount (6.60s)
	service_account_test.go:381: unexpected error: Operation cannot be fulfilled on serviceaccounts "deployer": the object has been modified; please apply your changes to the latest version and try again
FAIL
```

as seen here:
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_integration/1717/console

--- a/test/integration/service_account_test.go
+++ b/test/integration/service_account_test.go
@@ -376,7 +376,10 @@ func TestEnforcingServiceAccount(t *testing.T) {
 
 	time.Sleep(5)
 
-	_, err = clusterAdminKubeClient.ServiceAccounts(api.NamespaceDefault).Update(sa)
+	err = kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
+		_, err := clusterAdminKubeClient.ServiceAccounts(api.NamespaceDefault).Update(sa)
+		return err
+	})
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}

